v3.0.2 local html docs "git describe" error

<!--To help us understand and resolve your issue, please fill out the form to the best of your ability.-->
<!--You can feel free to delete the sections that do not apply.-->

### Bug report

**Bug summary**

<!--A short 1-2 sentences that succinctly describes the bug-->

*make html* fails when building html docs from downloaded source (https://github.com/matplotlib/matplotlib/archive/v3.0.2.zip), probably, because *.git* directory is not included there. The intention is to build docs for offline use.

**Code for reproduction**

<!--A minimum code snippet required to reproduce the bug, also minimizing the number of dependencies required-->

```bash
make html
```

**Actual outcome**

<!--The output produced by the above code, which may be a screenshot, console output, etc.-->

```
Running Sphinx v1.7.6
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

Configuration error:
There is a programable error in your configuration file:

Traceback (most recent call last):
  File "/home/<username>/.local/lib/python3.6/site-packages/sphinx/config.py", line 161, in __init__
    execfile_(filename, config)
  File "/home/<username>/.local/lib/python3.6/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
    exec_(code, _globals)
  File "conf.py", line 145, in <module>
    SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'describe', '--dirty']' returned non-zero exit status 128.

Makefile:20: recipe for target 'html' failed
make: *** [html] Error 2
```

**Expected outcome**

<!--A description of the expected outcome from the code snippet-->

A set of html docs.

<!--If this used to work in an earlier version of Matplotlib, please note the version it used to work on-->

**Matplotlib version**
<!--Please specify your platform and versions of the relevant libraries you are using:-->
  * Operating system: Ubuntu 18.04.1 LTS
  * Matplotlib version: 3.0.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): agg
  * Python version: Python 3.6.6
  * Jupyter version (if applicable):
  * Other libraries: 

<!--Please tell us how you installed matplotlib and python e.g., from source, pip, conda-->
<!--If you installed from conda, please specify which channel you used if not the default-->

**Workaround**

Modify *conf.py*:

```python
#SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
SHA = "8858a0d1bdd149a0897789e8503ac586be14676d"
```

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -11,6 +11,7 @@
 
 import os
 import shutil
+import subprocess
 import sys
 
 import matplotlib
@@ -141,8 +142,13 @@ source_encoding = "utf-8"
 master_doc = 'contents'
 
 # General substitutions.
-from subprocess import check_output
-SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
+try:
+    SHA = subprocess.check_output(
+        ['git', 'describe', '--dirty']).decode('utf-8').strip()
+# Catch the case where git is not installed locally, and use the versioneer
+# version number instead
+except subprocess.CalledProcessError:
+    SHA = matplotlib.__version__
 
 html_context = {'sha': SHA}
 

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -147,7 +147,7 @@ try:
         ['git', 'describe', '--dirty']).decode('utf-8').strip()
 # Catch the case where git is not installed locally, and use the versioneer
 # version number instead
-except subprocess.CalledProcessError:
+except (subprocess.CalledProcessError, FileNotFoundError):
     SHA = matplotlib.__version__
 
 html_context = {'sha': SHA}

